Dockerfile

```diff
@@ -3,32 +3,41 @@
 # Copyright Authors of Cilium
 # SPDX-License-Identifier: Apache-2.0
 
+# FINAL_CONTAINER specifies the source for the output
+# cilium-cli-ci (default) is based on ubuntu with cloud CLIs
+# cilium-cli is from scratch only including cilium binaries
+ARG FINAL_CONTAINER="cilium-cli-ci"
+
 FROM docker.io/library/golang:1.23.1-alpine3.19@sha256:e0ea2a119ae0939a6d449ea18b2b1ba30b44986ec48dbb88f3a93371b4bf8750 AS builder
 WORKDIR /go/src/github.com/cilium/cilium-cli
 RUN apk add --no-cache curl git make ca-certificates
 COPY . .
 RUN make
 
-FROM ubuntu:24.04@sha256:dfc10878be8d8fc9c61cbff33166cb1d1fe44391539243703c72766894fa834a
-LABEL maintainer="maintainer@cilium.io"
-WORKDIR /root/app
+FROM scratch AS cilium-cli
+COPY --from=builder /go/src/github.com/cilium/cilium-cli/cilium /usr/local/bin/cilium
+
+FROM ubuntu:24.04@sha256:dfc10878be8d8fc9c61cbff33166cb1d1fe44391539243703c72766894fa834a AS cilium-cli-ci
 COPY --from=builder /go/src/github.com/cilium/cilium-cli/cilium /usr/local/bin/cilium
```

Comment:
i guess we need to move this line to the final container? 💭

Reply:
In either scenario we get a resulting image layer, one built on top of ubuntu + tools + cilium
the other from scratch with just cilium

Reply:
That is because the ci pipelines define cilium as an argument instead of as the entrypoint
so i made it configurable for now so we can build with a correct entrypoint for the scratch container